tools/federation/src/test/java/demos/FedXWithRemoteRepositoryManager.java

```diff
@@ -0,0 +1,47 @@
+package demos;
+
+import org.eclipse.rdf4j.federated.FedXFactory;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.model.vocabulary.FOAF;
+import org.eclipse.rdf4j.model.vocabulary.RDF;
+import org.eclipse.rdf4j.repository.Repository;
+import org.eclipse.rdf4j.repository.RepositoryConnection;
+import org.eclipse.rdf4j.repository.RepositoryResult;
+import org.eclipse.rdf4j.repository.manager.RemoteRepositoryManager;
+import org.eclipse.rdf4j.repository.manager.RepositoryManager;
+
+/**
+ * Demo which shows an example of a local federation which uses repositories from a remote RDF4J instance as federation
+ * members.
+ * 
+ * @author Andreas Schwarte
+ *
+ */
+public class FedXWithRemoteRepositoryManager {
+
+	public static void main(String[] args) throws Exception {
+
+		// connection URL of a RDF4J server which manages the repositories
+		String serverUrl = "http://localhost:8080/rdf4j-server";
+		RepositoryManager repoManager = new RemoteRepositoryManager(serverUrl);
+
+		// assumes that remote repositories exist
+		Repository repo = FedXFactory.newFederation()
+				.withRepositoryResolver(repoManager)
+				.withResolvableEndpoint("my-repository-1")
+				.withResolvableEndpoint("my-repository-2")
+				.create();
+
+		repo.init();
+
+		try (RepositoryConnection conn = repo.getConnection()) {
+			try (RepositoryResult<Statement> repoResult = conn.getStatements(null, RDF.TYPE, FOAF.PERSON)) {
+				repoResult.forEach(st -> System.out.println(st));
+			}
+		}
+
+		repo.shutDown();
+		repoManager.shutDown();
```

Comment:
You can just do a `shutDown` on the `repoManager`, it will shut down the actual repository for you. And perhaps, if we're a bit paranoid, this should be in a `finally` clause.

Reply:
The FedX repository is actually a local repository, while only the federation members are managed in the remote repository manager (i.e. in the HTTP server). Thus the shutdown is required. I tried to make this a bit more explicit in the example by naming the variable "localRepo"

The finally for the shutdown I typically omit in the very simplistic examples. I typically only add the try with resources for connections and tuple results.

Reply:
Fair enough, makes sense.